src/main/java/bio/terra/cda/app/controller/QueryApiController.java

```diff
@@ -18,58 +20,79 @@
 @Controller
 public class QueryApiController implements QueryApi {
 
-  public static final int DEFAULT_LIMIT = 100;
-  public static final int DEFAULT_OFFSET = 0;
+  public static final int DEFAULT_PAGE_SIZE = 100;
+  public static final int DEFAULT_OFFSET = 100;
 
   private final QueryService queryService;
   private final ApplicationConfiguration applicationConfiguration;
+  private final HttpServletRequest webRequest;
 
   @Autowired
   public QueryApiController(
-      QueryService queryService, ApplicationConfiguration applicationConfiguration) {
+      QueryService queryService,
+      ApplicationConfiguration applicationConfiguration,
+      HttpServletRequest webRequest) {
     this.queryService = queryService;
     this.applicationConfiguration = applicationConfiguration;
+    this.webRequest = webRequest;
   }
 
-  private ResponseEntity<QueryResponseData> sendQuery(
-      @Valid String querySql, @Valid Integer offset, @Valid Integer limit, boolean dryRun) {
-    String queryStringWithPagination =
-        String.format(
-            "%s LIMIT %d OFFSET %d",
-            querySql,
-            Objects.requireNonNullElse(limit, DEFAULT_LIMIT),
-            Objects.requireNonNullElse(offset, DEFAULT_OFFSET));
+  private String createNextUrl(String jobId, int offset, int pageSize) {
+    var path = String.format("/api/v1/query/%s?offset=%s&pageSize=%s", jobId, offset, pageSize);
 
-    var result =
-        dryRun ? Collections.emptyList() : queryService.runQuery(queryStringWithPagination);
-    var response = new QueryResponseData().result(new ArrayList<>(result)).querySql(querySql);
+    try {
+      URL baseUrl = new URL(webRequest.getHeader("referer"));
+      return new URL(baseUrl.getProtocol(), baseUrl.getHost(), baseUrl.getPort(), path).toString();
+    } catch (MalformedURLException e) {
+      // Not sure what a good fallback would be here.
```

Comment:
You don't think throwing an exception here would make sense?

Reply:
Not really. At this point the query has generated a response with data in it. If we throw here, the data is lost. I don't think this will happen in practice , but I'm not 100% sure so for a fallback it just returns the path, which is better than nothing IMO.

Maybe there's a better way to do this? All I want is the URL the user used to get to this endpoint, I'm sure Spring has it somewhere.